src/utils/store-odp-data.js

```diff
@@ -0,0 +1,150 @@
+const redis = require('redis')
+const request = require('request-promise-native')
+const { promisify } = require('util')
+
+async function redisClientConnect() {
+  return redis.createClient({
+    port: process.env.REDIS_PORT,
+    host: process.env.REDIS_HOST,
+    password: process.env.REDIS_PASSWORD
+  })
+}
+
+async function getDatasets() {
+  let rawJsonData = await request(process.env.OPEN_DATA_URL)
+  const rawData = JSON.parse(rawJsonData)
+  return rawData.dataset.map(function(entry) {
+    return { label: entry.title, identifier: entry.identifier }
+  })
+}
+
+function sortOrder(a, b) {
+  return a.label < b.label ? -1 : a.label > b.label ? 1 : 0
+}
+
+async function getDatasetColumns(identifier) {
+  let rawJsonDataColumns = await request(identifier)
+  const rawDataColumns = JSON.parse(rawJsonDataColumns)
+  const columns = rawDataColumns.columns
+    .map(function(entry) {
+      return { label: entry.name, value: entry.id }
+    })
+    .sort(sortOrder)
+  return columns
+}
+
+function insert(client, key, dataset_label, values) {
+  const data = {
+    label: dataset_label,
+    values: values
+  }
+  client.set(key, JSON.stringify(data))
+}
+
+function insertCache(cache, key, dataset_label, columns) {
+  cache.insertDataset(key, dataset_label, columns)
+}
+
+async function storeData(client, dataset, isCache = false) {
+  const { label, identifier } = dataset
+  const key = isCache
+    ? identifier.slice(35)
+    : 'opendata:' + identifier.slice(35) /** unique identifier */
+  await getDatasetColumns(identifier)
+    .then(columns => {
+      isCache
```

Comment:
I'm not a huge fan of ternaries. I think there are cases where they make sense, but they're harming readability here.